{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/Xinrui-Fang/-Python-Crawler-Turorial-1-/blob/master/MotionGPT_colab.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "id": "VjYy0F2gZIPR",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "0c30e100-ccef-49f8-c663-90ae8189a904"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content\n",
            "fatal: destination path 'MotionGPT-hf' already exists and is not an empty directory.\n",
            "/content/MotionGPT-hf\n",
            "aria2 is already the newest version (1.36.0-1).\n",
            "libosmesa6 is already the newest version (23.2.1-1ubuntu3.1~22.04.2).\n",
            "0 upgraded, 0 newly installed, 0 to remove and 49 not upgraded.\n",
            "\n",
            "Download Results:\n",
            "gid   |stat|avg speed  |path/URI\n",
            "======+====+===========+=======================================================\n",
            "651f4d|\u001b[1;32mOK\u001b[0m  |       0B/s|/content/MotionGPT-hf/assets/meta/mean_eval.npy\n",
            "\n",
            "Status Legend:\n",
            "(OK):download completed.\n",
            "\n",
            "Download Results:\n",
            "gid   |stat|avg speed  |path/URI\n",
            "======+====+===========+=======================================================\n",
            "97825b|\u001b[1;32mOK\u001b[0m  |       0B/s|/content/MotionGPT-hf/assets/meta/mean.npy\n",
            "\n",
            "Status Legend:\n",
            "(OK):download completed.\n",
            "\n",
            "Download Results:\n",
            "gid   |stat|avg speed  |path/URI\n",
            "======+====+===========+=======================================================\n",
            "0c14d1|\u001b[1;32mOK\u001b[0m  |       0B/s|/content/MotionGPT-hf/assets/meta/std_eval.npy\n",
            "\n",
            "Status Legend:\n",
            "(OK):download completed.\n",
            "\n",
            "Download Results:\n",
            "gid   |stat|avg speed  |path/URI\n",
            "======+====+===========+=======================================================\n",
            "b566e5|\u001b[1;32mOK\u001b[0m  |       0B/s|/content/MotionGPT-hf/assets/meta/std.npy\n",
            "\n",
            "Status Legend:\n",
            "(OK):download completed.\n",
            "\n",
            "Download Results:\n",
            "gid   |stat|avg speed  |path/URI\n",
            "======+====+===========+=======================================================\n",
            "7a96d2|\u001b[1;32mOK\u001b[0m  |       0B/s|/content/MotionGPT-hf/assets/videos/example0_blender.mp4\n",
            "\n",
            "Status Legend:\n",
            "(OK):download completed.\n",
            "\n",
            "Download Results:\n",
            "gid   |stat|avg speed  |path/URI\n",
            "======+====+===========+=======================================================\n",
            "92a801|\u001b[1;32mOK\u001b[0m  |       0B/s|/content/MotionGPT-hf/deps/smpl/smpl_models/gmm_08.pkl\n",
            "\n",
            "Status Legend:\n",
            "(OK):download completed.\n",
            "\n",
            "Download Results:\n",
            "gid   |stat|avg speed  |path/URI\n",
            "======+====+===========+=======================================================\n",
            "0c5385|\u001b[1;32mOK\u001b[0m  |       0B/s|/content/MotionGPT-hf/deps/smpl/smpl_models/neutral_smpl_mean_params.h5\n",
            "\n",
            "Status Legend:\n",
            "(OK):download completed.\n",
            "\n",
            "Download Results:\n",
            "gid   |stat|avg speed  |path/URI\n",
            "======+====+===========+=======================================================\n",
            "88d9f4|\u001b[1;32mOK\u001b[0m  |       0B/s|/content/MotionGPT-hf/deps/smpl/smpl_models/SMPL_downsample_index.pkl\n",
            "\n",
            "Status Legend:\n",
            "(OK):download completed.\n",
            "\n",
            "Download Results:\n",
            "gid   |stat|avg speed  |path/URI\n",
            "======+====+===========+=======================================================\n",
            "10c3ed|\u001b[1;32mOK\u001b[0m  |       0B/s|/content/MotionGPT-hf/deps/smpl/smpl_models/smpl.tar.gz\n",
            "\n",
            "Status Legend:\n",
            "(OK):download completed.\n",
            "\n",
            "Download Results:\n",
            "gid   |stat|avg speed  |path/URI\n",
            "======+====+===========+=======================================================\n",
            "a7c0d7|\u001b[1;32mOK\u001b[0m  |       0B/s|/content/MotionGPT-hf/deps/smpl/smpl_models/smpl/SMPL_FEMALE.pkl\n",
            "\n",
            "Status Legend:\n",
            "(OK):download completed.\n",
            "\n",
            "Download Results:\n",
            "gid   |stat|avg speed  |path/URI\n",
            "======+====+===========+=======================================================\n",
            "6d43eb|\u001b[1;32mOK\u001b[0m  |       0B/s|/content/MotionGPT-hf/deps/smpl/smpl_models/smpl/SMPL_MALE.pkl\n",
            "\n",
            "Status Legend:\n",
            "(OK):download completed.\n",
            "\n",
            "Download Results:\n",
            "gid   |stat|avg speed  |path/URI\n",
            "======+====+===========+=======================================================\n",
            "c896ec|\u001b[1;32mOK\u001b[0m  |       0B/s|/content/MotionGPT-hf/deps/smpl/smpl_models/smpl/SMPL_NEUTRAL.pkl\n",
            "\n",
            "Status Legend:\n",
            "(OK):download completed.\n",
            "\n",
            "11/01 13:07:27 [\u001b[1;31mERROR\u001b[0m] CUID#7 - Download aborted. URI=https://huggingface.co/spaces/OpenMotionLab/MotionGPT/resolve/main/deps/smpl/smpl_models/smplh/mano_v1_2.zip\n",
            "Exception: [AbstractCommand.cc:351] errorCode=3 URI=https://huggingface.co/spaces/OpenMotionLab/MotionGPT/resolve/main/deps/smpl/smpl_models/smplh/mano_v1_2.zip\n",
            "  -> [HttpSkipResponseCommand.cc:218] errorCode=3 Resource not found\n",
            "\n",
            "Download Results:\n",
            "gid   |stat|avg speed  |path/URI\n",
            "======+====+===========+=======================================================\n",
            "ea7387|\u001b[1;31mERR\u001b[0m |       0B/s|/content/MotionGPT-hf/deps/smpl/smpl_models/smplh/mano_v1_2.zip\n",
            "\n",
            "Status Legend:\n",
            "(ERR):error occurred.\n",
            "\n",
            "aria2 will resume download if the transfer is restarted.\n",
            "If there are any errors, then see the log file. See '-l' option in help/man page for details.\n",
            "\n",
            "Download Results:\n",
            "gid   |stat|avg speed  |path/URI\n",
            "======+====+===========+=======================================================\n",
            "c5d6f3|\u001b[1;32mOK\u001b[0m  |       0B/s|/content/MotionGPT-hf/deps/smpl/smpl_models/smplh/SMPLH_FEMALE.npz\n",
            "\n",
            "Status Legend:\n",
            "(OK):download completed.\n",
            "\n",
            "Download Results:\n",
            "gid   |stat|avg speed  |path/URI\n",
            "======+====+===========+=======================================================\n",
            "8d176a|\u001b[1;32mOK\u001b[0m  |       0B/s|/content/MotionGPT-hf/deps/smpl/smpl_models/smplh/SMPLH_MALE.npz\n",
            "\n",
            "Status Legend:\n",
            "(OK):download completed.\n",
            "\n",
            "Download Results:\n",
            "gid   |stat|avg speed  |path/URI\n",
            "======+====+===========+=======================================================\n",
            "a97266|\u001b[1;32mOK\u001b[0m  |       0B/s|/content/MotionGPT-hf/deps/smpl/smpl_models/smplh/SMPLH_NEUTRAL.npz\n",
            "\n",
            "Status Legend:\n",
            "(OK):download completed.\n",
            "\n",
            "Download Results:\n",
            "gid   |stat|avg speed  |path/URI\n",
            "======+====+===========+=======================================================\n",
            "622b5f|\u001b[1;32mOK\u001b[0m  |       0B/s|/content/MotionGPT-hf/deps/smpl/smpl_models/smplh/smplh.tar.xz\n",
            "\n",
            "Status Legend:\n",
            "(OK):download completed.\n",
            "\n",
            "Download Results:\n",
            "gid   |stat|avg speed  |path/URI\n",
            "======+====+===========+=======================================================\n",
            "e76785|\u001b[1;32mOK\u001b[0m  |       0B/s|/content/MotionGPT-hf/deps/smpl/smpl_models/smplx_parts_segm.pkl\n",
            "\n",
            "Status Legend:\n",
            "(OK):download completed.\n",
            "Requirement already satisfied: numpy==1.21.6 in /usr/local/lib/python3.10/dist-packages (1.21.6)\n",
            "Requirement already satisfied: chumpy==0.70 in /usr/local/lib/python3.10/dist-packages (0.70)\n",
            "Requirement already satisfied: scipy>=0.13.0 in /usr/local/lib/python3.10/dist-packages (from chumpy==0.70) (1.11.4)\n",
            "Requirement already satisfied: six>=1.11.0 in /usr/local/lib/python3.10/dist-packages (from chumpy==0.70) (1.16.0)\n",
            "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
            "pyrender 0.1.45 requires PyOpenGL==3.1.0, but you have pyopengl 3.1.4 which is incompatible.\u001b[0m\u001b[31m\n",
            "\u001b[0mSeed set to 1234\n",
            "Fetching 3 files: 100% 3/3 [00:00<00:00, 40329.85it/s]\n",
            "/usr/local/lib/python3.10/dist-packages/transformers/tokenization_utils_base.py:1601: FutureWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be depracted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github.com/huggingface/transformers/issues/31884\n",
            "  warnings.warn(\n",
            "/content/MotionGPT-hf/app.py:35: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.\n",
            "  state_dict = torch.load(f'{model_path}/motiongpt_s3_h3d.tar',\n",
            "/content/MotionGPT-hf/app.py:479: GradioDeprecationWarning: 'scale' value should be an integer. Using 0.85 will cause issues.\n",
            "  with gr.Column(scale=0.85):\n",
            "/content/MotionGPT-hf/app.py:510: GradioDeprecationWarning: 'scale' value should be an integer. Using 0.15 will cause issues.\n",
            "  with gr.Column(scale=0.15, min_width=150):\n",
            "Running on local URL:  http://127.0.0.1:7860\n",
            "IMPORTANT: You are using gradio version 3.50.2, however version 4.44.1 is available, please upgrade.\n",
            "--------\n",
            "Running on public URL: https://dcc39c082786de8fad.gradio.live\n",
            "\n",
            "This share link expires in 72 hours. For free permanent hosting and GPU upgrades, run `gradio deploy` from Terminal to deploy to Spaces (https://huggingface.co/spaces)\n",
            "['A man is running']\n",
            "['<motion_id_512><motion_id_180><motion_id_180><motion_id_180><motion_id_180><motion_id_180><motion_id_72><motion_id_360><motion_id_25><motion_id_427><motion_id_467><motion_id_352><motion_id_170><motion_id_499><motion_id_263><motion_id_383><motion_id_93><motion_id_180><motion_id_55><motion_id_433><motion_id_373><motion_id_513>']\n",
            "error: XDG_RUNTIME_DIR not set in the environment.\n",
            "ALSA lib confmisc.c:855:(parse_card) cannot find card '0'\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_card_inum returned error: No such file or directory\n",
            "ALSA lib confmisc.c:422:(snd_func_concat) error evaluating strings\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory\n",
            "ALSA lib confmisc.c:1334:(snd_func_refer) error evaluating name\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory\n",
            "ALSA lib conf.c:5701:(snd_config_expand) Evaluate error: No such file or directory\n",
            "ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM default\n",
            "ALSA lib confmisc.c:855:(parse_card) cannot find card '0'\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_card_inum returned error: No such file or directory\n",
            "ALSA lib confmisc.c:422:(snd_func_concat) error evaluating strings\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory\n",
            "ALSA lib confmisc.c:1334:(snd_func_refer) error evaluating name\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory\n",
            "ALSA lib conf.c:5701:(snd_config_expand) Evaluate error: No such file or directory\n",
            "ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM default\n",
            "WARNING: You are using a SMPL model, with only 10 shape coefficients.\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/lib/python3.10/runpy.py\", line 196, in _run_module_as_main\n",
            "    return _run_code(code, main_globals, None,\n",
            "  File \"/usr/lib/python3.10/runpy.py\", line 86, in _run_code\n",
            "    exec(code, run_globals)\n",
            "  File \"/content/MotionGPT-hf/render.py\", line 59, in <module>\n",
            "    render.init_renderer([shape[0], shape[1], 3], params)\n",
            "  File \"/content/MotionGPT-hf/mGPT/render/pyrender/smpl_render.py\", line 138, in init_renderer\n",
            "    self.renderer = Renderer(vertices=self.vertices, focal_length=self.focal_length,\n",
            "  File \"/content/MotionGPT-hf/mGPT/render/pyrender/smpl_render.py\", line 24, in __init__\n",
            "    self.renderer = pyrender.OffscreenRenderer(viewport_width=img_res[0],\n",
            "  File \"/usr/local/lib/python3.10/dist-packages/pyrender/offscreen.py\", line 31, in __init__\n",
            "    self._create()\n",
            "  File \"/usr/local/lib/python3.10/dist-packages/pyrender/offscreen.py\", line 149, in _create\n",
            "    self._platform.init_context()\n",
            "  File \"/usr/local/lib/python3.10/dist-packages/pyrender/platforms/osmesa.py\", line 19, in init_context\n",
            "    from OpenGL.osmesa import (\n",
            "ImportError: cannot import name 'OSMesaCreateContextAttribs' from 'OpenGL.osmesa' (/usr/local/lib/python3.10/dist-packages/OpenGL/osmesa/__init__.py)\n",
            "Keyboard interruption in main thread... closing server.\n",
            "Killing tunnel 127.0.0.1:7860 <> https://dcc39c082786de8fad.gradio.live\n",
            "Keyboard interruption in main thread... closing server.\n",
            "^C\n"
          ]
        }
      ],
      "source": [
        "\n",
        "\n",
        "%cd /content\n",
        "!git clone -b dev https://github.com/camenduru/MotionGPT-hf\n",
        "%cd /content/MotionGPT-hf\n",
        "\n",
        "!apt -y install -qq aria2 libosmesa6\n",
        "\n",
        "!aria2c --console-log-level=error -c -x 16 -s 16 -k 1M https://huggingface.co/spaces/OpenMotionLab/MotionGPT/resolve/main/assets/meta/mean_eval.npy -d /content/MotionGPT-hf/assets/meta -o mean_eval.npy\n",
        "!aria2c --console-log-level=error -c -x 16 -s 16 -k 1M https://huggingface.co/spaces/OpenMotionLab/MotionGPT/resolve/main/assets/meta/mean.npy -d /content/MotionGPT-hf/assets/meta -o mean.npy\n",
        "!aria2c --console-log-level=error -c -x 16 -s 16 -k 1M https://huggingface.co/spaces/OpenMotionLab/MotionGPT/resolve/main/assets/meta/std_eval.npy -d /content/MotionGPT-hf/assets/meta -o std_eval.npy\n",
        "!aria2c --console-log-level=error -c -x 16 -s 16 -k 1M https://huggingface.co/spaces/OpenMotionLab/MotionGPT/resolve/main/assets/meta/std.npy -d /content/MotionGPT-hf/assets/meta -o std.npy\n",
        "!aria2c --console-log-level=error -c -x 16 -s 16 -k 1M https://huggingface.co/spaces/OpenMotionLab/MotionGPT/resolve/main/assets/videos/example0_blender.mp4 -d /content/MotionGPT-hf/assets/videos -o example0_blender.mp4\n",
        "!aria2c --console-log-level=error -c -x 16 -s 16 -k 1M https://huggingface.co/spaces/OpenMotionLab/MotionGPT/resolve/main/deps/smpl/smpl_models/gmm_08.pkl -d /content/MotionGPT-hf/deps/smpl/smpl_models -o gmm_08.pkl\n",
        "!aria2c --console-log-level=error -c -x 16 -s 16 -k 1M https://huggingface.co/spaces/OpenMotionLab/MotionGPT/resolve/main/deps/smpl/smpl_models/neutral_smpl_mean_params.h5 -d /content/MotionGPT-hf/deps/smpl/smpl_models -o neutral_smpl_mean_params.h5\n",
        "!aria2c --console-log-level=error -c -x 16 -s 16 -k 1M https://huggingface.co/spaces/OpenMotionLab/MotionGPT/resolve/main/deps/smpl/smpl_models/SMPL_downsample_index.pkl -d /content/MotionGPT-hf/deps/smpl/smpl_models -o SMPL_downsample_index.pkl\n",
        "!aria2c --console-log-level=error -c -x 16 -s 16 -k 1M https://huggingface.co/spaces/OpenMotionLab/MotionGPT/resolve/main/deps/smpl/smpl_models/smpl.tar.gz -d /content/MotionGPT-hf/deps/smpl/smpl_models -o smpl.tar.gz\n",
        "!aria2c --console-log-level=error -c -x 16 -s 16 -k 1M https://huggingface.co/spaces/OpenMotionLab/MotionGPT/resolve/main/deps/smpl/smpl_models/smpl/SMPL_FEMALE.pkl -d /content/MotionGPT-hf/deps/smpl/smpl_models/smpl -o SMPL_FEMALE.pkl\n",
        "!aria2c --console-log-level=error -c -x 16 -s 16 -k 1M https://huggingface.co/spaces/OpenMotionLab/MotionGPT/resolve/main/deps/smpl/smpl_models/smpl/SMPL_MALE.pkl -d /content/MotionGPT-hf/deps/smpl/smpl_models/smpl -o SMPL_MALE.pkl\n",
        "!aria2c --console-log-level=error -c -x 16 -s 16 -k 1M https://huggingface.co/spaces/OpenMotionLab/MotionGPT/resolve/main/deps/smpl/smpl_models/smpl/SMPL_NEUTRAL.pkl -d /content/MotionGPT-hf/deps/smpl/smpl_models/smpl -o SMPL_NEUTRAL.pkl\n",
        "!aria2c --console-log-level=error -c -x 16 -s 16 -k 1M https://huggingface.co/spaces/OpenMotionLab/MotionGPT/resolve/main/deps/smpl/smpl_models/smplh/mano_v1_2.zip -d /content/MotionGPT-hf/deps/smpl/smpl_models/smplh -o mano_v1_2.zip\n",
        "!aria2c --console-log-level=error -c -x 16 -s 16 -k 1M https://huggingface.co/spaces/OpenMotionLab/MotionGPT/resolve/main/deps/smpl/smpl_models/smplh/SMPLH_FEMALE.npz -d /content/MotionGPT-hf/deps/smpl/smpl_models/smplh -o SMPLH_FEMALE.npz\n",
        "!aria2c --console-log-level=error -c -x 16 -s 16 -k 1M https://huggingface.co/spaces/OpenMotionLab/MotionGPT/resolve/main/deps/smpl/smpl_models/smplh/SMPLH_MALE.npz -d /content/MotionGPT-hf/deps/smpl/smpl_models/smplh -o SMPLH_MALE.npz\n",
        "!aria2c --console-log-level=error -c -x 16 -s 16 -k 1M https://huggingface.co/spaces/OpenMotionLab/MotionGPT/resolve/main/deps/smpl/smpl_models/smplh/SMPLH_NEUTRAL.npz -d /content/MotionGPT-hf/deps/smpl/smpl_models/smplh -o SMPLH_NEUTRAL.npz\n",
        "!aria2c --console-log-level=error -c -x 16 -s 16 -k 1M https://huggingface.co/spaces/OpenMotionLab/MotionGPT/resolve/main/deps/smpl/smpl_models/smplh/smplh.tar.xz -d /content/MotionGPT-hf/deps/smpl/smpl_models/smplh -o smplh.tar.xz\n",
        "!aria2c --console-log-level=error -c -x 16 -s 16 -k 1M https://huggingface.co/spaces/OpenMotionLab/MotionGPT/resolve/main/deps/smpl/smpl_models/smplx_parts_segm.pkl -d /content/MotionGPT-hf/deps/smpl/smpl_models -o smplx_parts_segm.pkl\n",
        "\n",
        "import os\n",
        "os.environ.pop(\"XDG_RUNTIME_DIR\", None)\n",
        "# Install compatible versions of numpy and chumpy\n",
        "!pip install numpy==1.21.6 chumpy==0.70\n",
        "\n",
        "# Modify chumpy to remove deprecated imports from numpy\n",
        "!sed -i 's/from numpy import bool, int, float, complex, object, unicode, str, nan, inf/from numpy import nan, inf/g' /usr/local/lib/python3.10/dist-packages/chumpy/__init__.py\n",
        "\n",
        "!pip install -q gradio==3.50.2 pytorch-lightning omegaconf bert-score pyglet mapbox-earcut\n",
        "!pip install -q PyOpenGL_accelerate smplx trimesh triangle ftfy pyrender -U\n",
        "!pip install -q pyopengl==3.1.4 -U\n",
        "\n",
        "!pip install -q smplx==0.1.28 trimesh pyrender\n",
        "\n",
        "\n",
        "!python app.py"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install PyOpenGL==3.1.5 PyOpenGL-accelerate"
      ],
      "metadata": {
        "id": "bwBRQ_XEv2ov",
        "outputId": "3e87908f-ff14-4073-9526-1c665094efb9",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "execution_count": 4,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting PyOpenGL==3.1.5\n",
            "  Downloading PyOpenGL-3.1.5-py3-none-any.whl.metadata (3.2 kB)\n",
            "Requirement already satisfied: PyOpenGL-accelerate in /usr/local/lib/python3.10/dist-packages (3.1.7)\n",
            "Downloading PyOpenGL-3.1.5-py3-none-any.whl (2.4 MB)\n",
            "\u001b[?25l   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m0.0/2.4 MB\u001b[0m \u001b[31m?\u001b[0m eta \u001b[36m-:--:--\u001b[0m\r\u001b[2K   \u001b[91m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[91m╸\u001b[0m \u001b[32m2.4/2.4 MB\u001b[0m \u001b[31m86.2 MB/s\u001b[0m eta \u001b[36m0:00:01\u001b[0m\r\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m2.4/2.4 MB\u001b[0m \u001b[31m45.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: PyOpenGL\n",
            "  Attempting uninstall: PyOpenGL\n",
            "    Found existing installation: PyOpenGL 3.1.0\n",
            "    Uninstalling PyOpenGL-3.1.0:\n",
            "      Successfully uninstalled PyOpenGL-3.1.0\n",
            "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
            "pyrender 0.1.45 requires PyOpenGL==3.1.0, but you have pyopengl 3.1.5 which is incompatible.\u001b[0m\u001b[31m\n",
            "\u001b[0mSuccessfully installed PyOpenGL-3.1.5\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!apt-get install -y libosmesa6-dev\n"
      ],
      "metadata": {
        "id": "-7tUbnD1xh9V",
        "outputId": "1e0b2655-4598-4996-b95d-c45c30268ab8",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "execution_count": 5,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  libdrm-dev libgl-dev libglx-dev libpciaccess-dev mesa-common-dev\n",
            "The following NEW packages will be installed:\n",
            "  libdrm-dev libgl-dev libglx-dev libosmesa6-dev libpciaccess-dev mesa-common-dev\n",
            "0 upgraded, 6 newly installed, 0 to remove and 49 not upgraded.\n",
            "Need to get 2,646 kB of archives.\n",
            "After this operation, 5,899 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/main amd64 libpciaccess-dev amd64 0.16-3 [21.9 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libdrm-dev amd64 2.4.113-2~ubuntu0.22.04.1 [292 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy/main amd64 libglx-dev amd64 1.4.0-1 [14.1 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu jammy/main amd64 libgl-dev amd64 1.4.0-1 [101 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 mesa-common-dev amd64 23.2.1-1ubuntu3.1~22.04.2 [2,208 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libosmesa6-dev amd64 23.2.1-1ubuntu3.1~22.04.2 [8,984 B]\n",
            "Fetched 2,646 kB in 2s (1,505 kB/s)\n",
            "Selecting previously unselected package libpciaccess-dev:amd64.\n",
            "(Reading database ... 123707 files and directories currently installed.)\n",
            "Preparing to unpack .../0-libpciaccess-dev_0.16-3_amd64.deb ...\n",
            "Unpacking libpciaccess-dev:amd64 (0.16-3) ...\n",
            "Selecting previously unselected package libdrm-dev:amd64.\n",
            "Preparing to unpack .../1-libdrm-dev_2.4.113-2~ubuntu0.22.04.1_amd64.deb ...\n",
            "Unpacking libdrm-dev:amd64 (2.4.113-2~ubuntu0.22.04.1) ...\n",
            "Selecting previously unselected package libglx-dev:amd64.\n",
            "Preparing to unpack .../2-libglx-dev_1.4.0-1_amd64.deb ...\n",
            "Unpacking libglx-dev:amd64 (1.4.0-1) ...\n",
            "Selecting previously unselected package libgl-dev:amd64.\n",
            "Preparing to unpack .../3-libgl-dev_1.4.0-1_amd64.deb ...\n",
            "Unpacking libgl-dev:amd64 (1.4.0-1) ...\n",
            "Selecting previously unselected package mesa-common-dev:amd64.\n",
            "Preparing to unpack .../4-mesa-common-dev_23.2.1-1ubuntu3.1~22.04.2_amd64.deb ...\n",
            "Unpacking mesa-common-dev:amd64 (23.2.1-1ubuntu3.1~22.04.2) ...\n",
            "Selecting previously unselected package libosmesa6-dev:amd64.\n",
            "Preparing to unpack .../5-libosmesa6-dev_23.2.1-1ubuntu3.1~22.04.2_amd64.deb ...\n",
            "Unpacking libosmesa6-dev:amd64 (23.2.1-1ubuntu3.1~22.04.2) ...\n",
            "Setting up libpciaccess-dev:amd64 (0.16-3) ...\n",
            "Setting up libglx-dev:amd64 (1.4.0-1) ...\n",
            "Setting up libgl-dev:amd64 (1.4.0-1) ...\n",
            "Setting up libdrm-dev:amd64 (2.4.113-2~ubuntu0.22.04.1) ...\n",
            "Setting up libosmesa6-dev:amd64 (23.2.1-1ubuntu3.1~22.04.2) ...\n",
            "Setting up mesa-common-dev:amd64 (23.2.1-1ubuntu3.1~22.04.2) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import os\n",
        "os.environ[\"PYOPENGL_PLATFORM\"] = \"osmesa\""
      ],
      "metadata": {
        "id": "z-f3cxd00kB4"
      },
      "execution_count": 6,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!python app.py\n"
      ],
      "metadata": {
        "id": "JMnPohvA0wU6",
        "outputId": "849bc53e-9c4e-434a-9758-97de90076913",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Seed set to 1234\n",
            "Fetching 3 files: 100% 3/3 [00:00<00:00, 38245.93it/s]\n",
            "/usr/local/lib/python3.10/dist-packages/transformers/tokenization_utils_base.py:1601: FutureWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be depracted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github.com/huggingface/transformers/issues/31884\n",
            "  warnings.warn(\n",
            "/content/MotionGPT-hf/app.py:35: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.\n",
            "  state_dict = torch.load(f'{model_path}/motiongpt_s3_h3d.tar',\n",
            "/content/MotionGPT-hf/app.py:479: GradioDeprecationWarning: 'scale' value should be an integer. Using 0.85 will cause issues.\n",
            "  with gr.Column(scale=0.85):\n",
            "/content/MotionGPT-hf/app.py:510: GradioDeprecationWarning: 'scale' value should be an integer. Using 0.15 will cause issues.\n",
            "  with gr.Column(scale=0.15, min_width=150):\n",
            "Running on local URL:  http://127.0.0.1:7860\n",
            "IMPORTANT: You are using gradio version 3.50.2, however version 4.44.1 is available, please upgrade.\n",
            "--------\n",
            "Running on public URL: https://18bbfb074c8dbd93a3.gradio.live\n",
            "\n",
            "This share link expires in 72 hours. For free permanent hosting and GPU upgrades, run `gradio deploy` from Terminal to deploy to Spaces (https://huggingface.co/spaces)\n",
            "['the man is running']\n",
            "['<motion_id_512><motion_id_28><motion_id_28><motion_id_28><motion_id_28><motion_id_28><motion_id_28><motion_id_28><motion_id_313><motion_id_170><motion_id_491><motion_id_78><motion_id_25><motion_id_500><motion_id_467><motion_id_352><motion_id_439><motion_id_383><motion_id_383><motion_id_28><motion_id_28><motion_id_28><motion_id_290><motion_id_383><motion_id_513>']\n",
            "error: XDG_RUNTIME_DIR not set in the environment.\n",
            "ALSA lib confmisc.c:855:(parse_card) cannot find card '0'\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_card_inum returned error: No such file or directory\n",
            "ALSA lib confmisc.c:422:(snd_func_concat) error evaluating strings\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory\n",
            "ALSA lib confmisc.c:1334:(snd_func_refer) error evaluating name\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory\n",
            "ALSA lib conf.c:5701:(snd_config_expand) Evaluate error: No such file or directory\n",
            "ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM default\n",
            "ALSA lib confmisc.c:855:(parse_card) cannot find card '0'\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_card_inum returned error: No such file or directory\n",
            "ALSA lib confmisc.c:422:(snd_func_concat) error evaluating strings\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory\n",
            "ALSA lib confmisc.c:1334:(snd_func_refer) error evaluating name\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory\n",
            "ALSA lib conf.c:5701:(snd_config_expand) Evaluate error: No such file or directory\n",
            "ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM default\n",
            "WARNING: You are using a SMPL model, with only 10 shape coefficients.\n",
            "Moviepy - Building video cache/2024-11-01-13_13_4268067.mp4.\n",
            "Moviepy - Writing video cache/2024-11-01-13_13_4268067.mp4\n",
            "\n",
            "Moviepy - Done !\n",
            "Moviepy - video ready cache/2024-11-01-13_13_4268067.mp4\n",
            "['The man is waving to me']\n",
            "['<motion_id_512><motion_id_256><motion_id_423><motion_id_256><motion_id_369><motion_id_146><motion_id_146><motion_id_403><motion_id_403><motion_id_403><motion_id_91><motion_id_14><motion_id_281><motion_id_423><motion_id_513>']\n",
            "error: XDG_RUNTIME_DIR not set in the environment.\n",
            "ALSA lib confmisc.c:855:(parse_card) cannot find card '0'\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_card_inum returned error: No such file or directory\n",
            "ALSA lib confmisc.c:422:(snd_func_concat) error evaluating strings\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory\n",
            "ALSA lib confmisc.c:1334:(snd_func_refer) error evaluating name\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory\n",
            "ALSA lib conf.c:5701:(snd_config_expand) Evaluate error: No such file or directory\n",
            "ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM default\n",
            "ALSA lib confmisc.c:855:(parse_card) cannot find card '0'\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_card_inum returned error: No such file or directory\n",
            "ALSA lib confmisc.c:422:(snd_func_concat) error evaluating strings\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory\n",
            "ALSA lib confmisc.c:1334:(snd_func_refer) error evaluating name\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory\n",
            "ALSA lib conf.c:5701:(snd_config_expand) Evaluate error: No such file or directory\n",
            "ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM default\n",
            "WARNING: You are using a SMPL model, with only 10 shape coefficients.\n",
            "Moviepy - Building video cache/2024-11-01-13_15_1344086.mp4.\n",
            "Moviepy - Writing video cache/2024-11-01-13_15_1344086.mp4\n",
            "\n",
            "Moviepy - Done !\n",
            "Moviepy - video ready cache/2024-11-01-13_15_1344086.mp4\n",
            "['bowing to the author']\n",
            "['<motion_id_512><motion_id_417><motion_id_300><motion_id_300><motion_id_93><motion_id_436><motion_id_406><motion_id_406><motion_id_406><motion_id_406><motion_id_406><motion_id_459><motion_id_240><motion_id_240><motion_id_240><motion_id_417><motion_id_417><motion_id_417><motion_id_256><motion_id_256><motion_id_417><motion_id_417><motion_id_417><motion_id_513>']\n",
            "error: XDG_RUNTIME_DIR not set in the environment.\n",
            "ALSA lib confmisc.c:855:(parse_card) cannot find card '0'\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_card_inum returned error: No such file or directory\n",
            "ALSA lib confmisc.c:422:(snd_func_concat) error evaluating strings\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory\n",
            "ALSA lib confmisc.c:1334:(snd_func_refer) error evaluating name\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory\n",
            "ALSA lib conf.c:5701:(snd_config_expand) Evaluate error: No such file or directory\n",
            "ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM default\n",
            "ALSA lib confmisc.c:855:(parse_card) cannot find card '0'\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_card_inum returned error: No such file or directory\n",
            "ALSA lib confmisc.c:422:(snd_func_concat) error evaluating strings\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory\n",
            "ALSA lib confmisc.c:1334:(snd_func_refer) error evaluating name\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory\n",
            "ALSA lib conf.c:5701:(snd_config_expand) Evaluate error: No such file or directory\n",
            "ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM default\n",
            "WARNING: You are using a SMPL model, with only 10 shape coefficients.\n",
            "Moviepy - Building video cache/2024-11-01-13_22_3870620.mp4.\n",
            "Moviepy - Writing video cache/2024-11-01-13_22_3870620.mp4\n",
            "\n",
            "Moviepy - Done !\n",
            "Moviepy - video ready cache/2024-11-01-13_22_3870620.mp4\n",
            "['the man returns back']\n",
            "['<motion_id_512><motion_id_315><motion_id_447><motion_id_350><motion_id_259><motion_id_86><motion_id_127><motion_id_511><motion_id_72><motion_id_202><motion_id_47><motion_id_148><motion_id_264><motion_id_264><motion_id_264><motion_id_336><motion_id_420><motion_id_336><motion_id_189><motion_id_69><motion_id_510><motion_id_28><motion_id_127><motion_id_336><motion_id_264><motion_id_336><motion_id_306><motion_id_373><motion_id_373><motion_id_127><motion_id_336><motion_id_119><motion_id_375><motion_id_368><motion_id_264><motion_id_119><motion_id_264><motion_id_504><motion_id_250><motion_id_21><motion_id_450><motion_id_450><motion_id_43><motion_id_140><motion_id_140><motion_id_140><motion_id_260><motion_id_140><motion_id_135><motion_id_513>']\n",
            "error: XDG_RUNTIME_DIR not set in the environment.\n",
            "ALSA lib confmisc.c:855:(parse_card) cannot find card '0'\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_card_inum returned error: No such file or directory\n",
            "ALSA lib confmisc.c:422:(snd_func_concat) error evaluating strings\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory\n",
            "ALSA lib confmisc.c:1334:(snd_func_refer) error evaluating name\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory\n",
            "ALSA lib conf.c:5701:(snd_config_expand) Evaluate error: No such file or directory\n",
            "ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM default\n",
            "ALSA lib confmisc.c:855:(parse_card) cannot find card '0'\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_card_inum returned error: No such file or directory\n",
            "ALSA lib confmisc.c:422:(snd_func_concat) error evaluating strings\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory\n",
            "ALSA lib confmisc.c:1334:(snd_func_refer) error evaluating name\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory\n",
            "ALSA lib conf.c:5701:(snd_config_expand) Evaluate error: No such file or directory\n",
            "ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM default\n",
            "WARNING: You are using a SMPL model, with only 10 shape coefficients.\n",
            "Moviepy - Building video cache/2024-11-01-13_39_4799460.mp4.\n",
            "Moviepy - Writing video cache/2024-11-01-13_39_4799460.mp4\n",
            "\n",
            "Moviepy - Done !\n",
            "Moviepy - video ready cache/2024-11-01-13_39_4799460.mp4\n",
            "['the man is running']\n",
            "['<motion_id_512><motion_id_325><motion_id_28><motion_id_28><motion_id_28><motion_id_505><motion_id_28><motion_id_326><motion_id_263><motion_id_192><motion_id_460><motion_id_395><motion_id_431><motion_id_176><motion_id_44><motion_id_39><motion_id_290><motion_id_395><motion_id_24><motion_id_24><motion_id_89><motion_id_359><motion_id_10><motion_id_128><motion_id_270><motion_id_367><motion_id_367><motion_id_128><motion_id_367><motion_id_513>']\n",
            "error: XDG_RUNTIME_DIR not set in the environment.\n",
            "ALSA lib confmisc.c:855:(parse_card) cannot find card '0'\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_card_inum returned error: No such file or directory\n",
            "ALSA lib confmisc.c:422:(snd_func_concat) error evaluating strings\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory\n",
            "ALSA lib confmisc.c:1334:(snd_func_refer) error evaluating name\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory\n",
            "ALSA lib conf.c:5701:(snd_config_expand) Evaluate error: No such file or directory\n",
            "ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM default\n",
            "ALSA lib confmisc.c:855:(parse_card) cannot find card '0'\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_card_inum returned error: No such file or directory\n",
            "ALSA lib confmisc.c:422:(snd_func_concat) error evaluating strings\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory\n",
            "ALSA lib confmisc.c:1334:(snd_func_refer) error evaluating name\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory\n",
            "ALSA lib conf.c:5701:(snd_config_expand) Evaluate error: No such file or directory\n",
            "ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM default\n",
            "WARNING: You are using a SMPL model, with only 10 shape coefficients.\n",
            "Moviepy - Building video cache/2024-11-01-13_40_5792584.mp4.\n",
            "Moviepy - Writing video cache/2024-11-01-13_40_5792584.mp4\n",
            "\n",
            "Moviepy - Done !\n",
            "Moviepy - video ready cache/2024-11-01-13_40_5792584.mp4\n",
            "['the man is running']\n",
            "['<motion_id_512><motion_id_28><motion_id_28><motion_id_28><motion_id_28><motion_id_28><motion_id_28><motion_id_28><motion_id_313><motion_id_170><motion_id_491><motion_id_78><motion_id_25><motion_id_500><motion_id_467><motion_id_352><motion_id_439><motion_id_383><motion_id_383><motion_id_28><motion_id_28><motion_id_28><motion_id_290><motion_id_383><motion_id_513>']\n",
            "error: XDG_RUNTIME_DIR not set in the environment.\n",
            "ALSA lib confmisc.c:855:(parse_card) cannot find card '0'\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_card_inum returned error: No such file or directory\n",
            "ALSA lib confmisc.c:422:(snd_func_concat) error evaluating strings\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory\n",
            "ALSA lib confmisc.c:1334:(snd_func_refer) error evaluating name\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory\n",
            "ALSA lib conf.c:5701:(snd_config_expand) Evaluate error: No such file or directory\n",
            "ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM default\n",
            "ALSA lib confmisc.c:855:(parse_card) cannot find card '0'\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_card_inum returned error: No such file or directory\n",
            "ALSA lib confmisc.c:422:(snd_func_concat) error evaluating strings\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory\n",
            "ALSA lib confmisc.c:1334:(snd_func_refer) error evaluating name\n",
            "ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory\n",
            "ALSA lib conf.c:5701:(snd_config_expand) Evaluate error: No such file or directory\n",
            "ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM default\n",
            "WARNING: You are using a SMPL model, with only 10 shape coefficients.\n",
            "Moviepy - Building video cache/2024-11-01-13_42_0442399.mp4.\n",
            "Moviepy - Writing video cache/2024-11-01-13_42_0442399.mp4\n",
            "\n",
            "Moviepy - Done !\n",
            "Moviepy - video ready cache/2024-11-01-13_42_0442399.mp4\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "40Vo7kKc0zYr"
      },
      "execution_count": null,
      "outputs": []
    }
  ],
  "metadata": {
    "accelerator": "GPU",
    "colab": {
      "gpuType": "T4",
      "provenance": [],
      "include_colab_link": true
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}